server/src/main/java/com/pepperkick/ems/server/service/DesignationService.java

```diff
@@ -0,0 +1,193 @@
+package com.pepperkick.ems.server.service;
+
+import com.pepperkick.ems.server.repository.DesignationRepository;
+import com.pepperkick.ems.server.repository.EmployeeRepository;
+import com.pepperkick.ems.server.requestbody.DesignationRequestPostBody;
+import com.pepperkick.ems.server.entity.Designation;
+import com.pepperkick.ems.server.entity.Employee;
+import com.pepperkick.ems.server.exception.BadRequestException;
+import com.pepperkick.ems.server.exception.NotFoundException;
+import com.pepperkick.ems.server.util.MessageHelper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.DataIntegrityViolationException;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Service
+public class DesignationService {
+    private final DesignationRepository designationRepository;
+    private final EmployeeRepository employeeRepository;
+    private final MessageHelper messageHelper;
+
+    @Autowired
+    public DesignationService(DesignationRepository designationRepository, EmployeeRepository employeeRepository, MessageHelper messageHelper) {
+        this.designationRepository = designationRepository;
+        this.employeeRepository = employeeRepository;
+        this.messageHelper = messageHelper;
+
+        // Check if designation table is empty, if yes then fill with initial data
+        List<Designation> designations = designationRepository.findAll();
+        if (designations.isEmpty()) {
+            String[] titles = { "Director", "Manager", "Lead", "Developer", "DevOps", "QA", "Intern" };
+            float[] levels = { 1, 2, 3, 4, 4, 4, 5 };
+
+            for (int i = 0; i < titles.length; i++) {
+                Designation designation = new Designation();
+                designation.setTitle(titles[i]);
+                designation.setLevel(levels[i]);
+                designationRepository.save(designation);
+            }
+        }
+    }
+
+    // Get designation level that is between two designations
+    public float getNewDesignationLevel(Designation higherDesignation) {
+        boolean flag = false;
+        Designation highest = higherDesignation;
+
+        // Get list of all designations ordered by level
+        List<Designation> designations = designationRepository.findAllByOrderByLevelAsc();
+
+        // Get index of highest designation in the list
+        int index = designations.indexOf(higherDesignation);
+
+        // Loop through the list to find the next highest designation by level
+        while (!flag) {
+            try {
+                Designation temp = designations.get(index++);
+
+                if (temp.getLevel() > highest.getLevel()) {
+                    highest = temp;
+                    flag = true;
+                }
+            } catch (IndexOutOfBoundsException e) {
+                break;
```

Comment:
catching exceptions like IndexOutOfBounds, NPE etc are considered bad practice unless you wrap them and throw as a new exception. Exceptions are not a good way to handle control flow in most cases